'''
Author: Shloka Negi, shnegi@ucsc.edu
Usage: Execute line by line, or blocks of code as required
Purpose: Haplotype comparisons (or benchmarking analysis) between giraffe_PE v/s Truth, giraffe_noPE v/s Truth, and BWA v/s Truth
Input file requirements: Query VCF file, Truth VCF file, Truth BED file, and Target BED file
Tool installation: Used installation from Adam Novak - anovak@ucsc.edu

Query Data:
Giraffe-noPE VCF (generated by google): https://storage.googleapis.com/brain-genomics/awcarroll/share/ucsc/vg_exomes/vcf/HG003.agilent.abra.vg.vg_exome_model.vcf.gz
BWA VCF (sent by google): https://storage.googleapis.com/brain-genomics/awcarroll/share/ucsc/vg_exomes/vcf/HG003.novaseq.wes_agilent.100x.dedup.bwa.vcf.gz
Giraffe-PE VCF (generated by us using Giraffe-DeepVariant workflow on Terra)

Truth Data:
HG003 Truth VCF: https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/release/AshkenazimTrio/HG003_NA24149_father/latest/GRCh38/HG003_GRCh38_1_22_v4.2.1_benchmark.vcf.gz
HG003 Truth BED for confident regions: https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/release/AshkenazimTrio/HG003_NA24149_father/latest/GRCh38/HG003_GRCh38_1_22_v4.2.1_benchmark_noinconsistent.bed

Target Data:
Agilent BED: https://storage.googleapis.com/brain-genomics-public/research/sequencing/grch38/bed/agilent.targets.grch38.bed
RefSeq BED: https://storage.googleapis.com/brain-genomics-public/research/sequencing/grch38/bed/grch38_refseq.bed
'''

### Working dir
SHNEGI="/public/groups/cgl/graph-genomes/shnegi"
### Intitialize directories
HAPPY="/public/home/anovak/build/hap.py/install"
INPUTS="/public/groups/cgl/graph-genomes/shnegi/inputs"
ref_genomes="/public/groups/cgl/graph-genomes/shnegi/ref-genomes"

## Run happy, with modes to Subset the kit specific target regions (-T) and providing confident regions (-f)
############# Agilent-kit Capture Region 
##1. For Giraffe_PE
${HAPPY}/bin/hap.py \
    ${INPUTS}/happy/HG003_GRCh38_1_22_v4.2.1_benchmark.vcf \
    ${INPUTS}/happy/vg.HG003.sorted.vcf \
    -f ${INPUTS}/happy/HG003_GRCh38_1_22_v4.2.1_benchmark_noinconsistent.bed \
    -T ${INPUTS}/happy/agilent.targets.grch38.bed \
    --reference ${ref_genomes}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna \
    -o ${SHNEGI}/happyout/vg_rerun/giraffePE_truth

##2. For Giraffe_noPE
${HAPPY}/bin/hap.py \
    ${INPUTS}/happy/HG003_GRCh38_1_22_v4.2.1_benchmark.vcf \
    ${INPUTS}/happy/HG003.agilent.abra.vg.vg_exome_model.vcf \
    -f ${INPUTS}/happy/HG003_GRCh38_1_22_v4.2.1_benchmark_noinconsistent.bed \
    -T ${INPUTS}/happy/agilent.targets.grch38.bed \
    --reference ${ref_genomes}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna \
    -o ${SHNEGI}/happyout/vg_rerun/giraffenoPE_truth

##3. For BWA
${HAPPY}/bin/hap.py \
    ${INPUTS}/happy/HG003_GRCh38_1_22_v4.2.1_benchmark.vcf \
    ${INPUTS}/happy/HG003.novaseq.wes_agilent.100x.dedup.bwa.vcf \
    -f ${INPUTS}/happy/HG003_GRCh38_1_22_v4.2.1_benchmark_noinconsistent.bed \
    -T ${INPUTS}/happy/agilent.targets.grch38.bed \
    --reference ${ref_genomes}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna \
    -o ${SHNEGI}/happyout/vg_rerun/bwa_truth


############# Ref-Seq Target Region 
##1. For Giraffe_PE
${HAPPY}/bin/hap.py \
    ${INPUTS}/happy/HG003_GRCh38_1_22_v4.2.1_benchmark.vcf \
    ${INPUTS}/happy/vg.HG003.sorted.vcf \
    -f ${INPUTS}/happy/HG003_GRCh38_1_22_v4.2.1_benchmark_noinconsistent.bed \
    -T ${INPUTS}/happy/grch38_refseq.bed \
    --reference ${ref_genomes}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna \
    -o ${SHNEGI}/happyout/vg_rerun/giraffePE_truth

##2. For Giraffe_noPE
${HAPPY}/bin/hap.py \
    ${INPUTS}/happy/HG003_GRCh38_1_22_v4.2.1_benchmark.vcf \
    ${INPUTS}/happy/HG003.agilent.abra.vg.vg_exome_model.vcf \
    -f ${INPUTS}/happy/HG003_GRCh38_1_22_v4.2.1_benchmark_noinconsistent.bed \
    -T ${INPUTS}/happy/grch38_refseq.bed \
    --reference ${ref_genomes}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna \
    -o ${SHNEGI}/happyout/vg_rerun/giraffenoPE_truth

##3. For BWA
${HAPPY}/bin/hap.py \
    ${INPUTS}/happy/HG003_GRCh38_1_22_v4.2.1_benchmark.vcf \
    ${INPUTS}/happy/HG003.novaseq.wes_agilent.100x.dedup.bwa.vcf \
    -f ${INPUTS}/happy/HG003_GRCh38_1_22_v4.2.1_benchmark_noinconsistent.bed \
    -T ${INPUTS}/happy/grch38_refseq.bed \
    --reference ${ref_genomes}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna \
    -o ${SHNEGI}/happyout/vg_rerun/bwa_truth
